src/engine/IndexScan.cpp

```diff
@@ -247,56 +241,50 @@ size_t IndexScan::computeSizeEstimate() {
 // _____________________________________________________________________________
 void IndexScan::computeSPOfreeP(ResultTable* result) const {
   result->_nofColumns = 2;
+  result->_data.setCols(result->_nofColumns);
   result->_resultTypes.push_back(ResultTable::ResultType::KB);
   result->_resultTypes.push_back(ResultTable::ResultType::KB);
   result->_sortedBy = {0, 1};
-  result->_fixedSizeData = new vector<array<Id, 2>>();
-  _executionContext->getIndex().scanSPO(
-      _subject, static_cast<vector<array<Id, 2>>*>(result->_fixedSizeData));
+  _executionContext->getIndex().scanSPO(_subject, &result->_data);
 }
 
 // _____________________________________________________________________________
 void IndexScan::computeSOPboundO(ResultTable* result) const {
   result->_nofColumns = 1;
+  result->_data.setCols(result->_nofColumns);
   result->_resultTypes.push_back(ResultTable::ResultType::KB);
   result->_sortedBy = {0};
-  result->_fixedSizeData = new vector<array<Id, 1>>();
-  _executionContext->getIndex().scanSOP(
-      _subject, _object,
-      static_cast<vector<array<Id, 1>>*>(result->_fixedSizeData));
+  _executionContext->getIndex().scanSOP(_subject, _object, &result->_data);
 }
 
 // _____________________________________________________________________________
 void IndexScan::computeSOPfreeO(ResultTable* result) const {
   result->_nofColumns = 2;
+  result->_data.setCols(result->_nofColumns);
```

Comment:
I think we could remove the `_nofColumns` member from `ResultTable` that would only save 8 bytes but also remove the line above this line

Reply:
Removed